### ISSUE TITLE
About lm_loss for distillm

### ISSUE THREAD

**Issue by songmzhang** [jongwooko/distillm#16]
Hi authors, I found that there may be a problem in your code when calculating lm_loss for distillm. As shown in the following code, lm_loss is calculated with logits and labels in `no_model_batch`. However, sometimes `model_batch` and `no_model_batch` are from **SGO**, and calculating lm_loss on this data will push the student model to learn the outputs generated by itself. 
https://github.com/jongwooko/distillm/blob/d47e77ff9d27721783b32213be38c1204230cc0a/finetune.py#L321-L339

In our experiments, we found this will lead to model collapse when the threshold increases (after several epochs).
